app/src/main/java/com/android/unio/ui/association/AssociationProfile.kt

```diff
@@ -11,13 +11,28 @@ import androidx.compose.material3.Text
 import androidx.compose.material3.TopAppBar
 import androidx.compose.runtime.Composable
 import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.Color
 import androidx.compose.ui.platform.testTag
+import androidx.lifecycle.viewmodel.compose.viewModel
+import com.android.unio.model.association.AssociationViewModel
 import com.android.unio.ui.navigation.NavigationAction
+import com.android.unio.ui.theme.AppTypography
 
 @OptIn(ExperimentalMaterial3Api::class)
 @SuppressLint("UnusedMaterial3ScaffoldPaddingParameter")
 @Composable
-fun AssociationProfile(navigationAction: NavigationAction) {
+fun AssociationProfile(
+    navigationAction: NavigationAction,
+    associationId: String,
+    associationViewModel: AssociationViewModel = viewModel(factory = AssociationViewModel.Factory)
+) {
+  val association =
+      associationViewModel.findAssociationById(associationId)
+          ?: return Text(
```

Comment:
Good error handling, maybe this could even be split in an `if` statement for code clarity and an additional `Log.e` call.